### PR TITLE
[Python code]: fix leak of validation examples into training loop

### PR DESCRIPTION
**Pull request by KellerJordan**
I was running this code with validation enabled every 128 steps when I noticed a strange pattern in the loss:

Screenshot 2024-05-30 at 10 41 34 AM

This is caused by the fact that the validation batches share a scope and name with the training batches. So, the network ends up training on the last batch of validation examples once after every run of validation.

This can be fixed by giving the valid batches a different name. I also gave the batches used for sampling a different name.

Thanks

### CODE REVIEW

**Comment by karpathy** [comment]
I diverged from the way train_gpt2.cu iterates because I had to save the first iteration's data to .bin files, but now we get this bug because of it. I don't actually love this solution because there is a deeper issue here. Basically I want to mirror the logic as it is in the train_gpt2.cu file. Let me see

**Comment by karpathy** [comment]
I pushed a fix to master, have a look. closing.
